### PR TITLE
Add worker queue support to job class and model

### LINKED ISSUE #2064: Job Queue Routing: Add Job Class Definition for Preferred Queue(s)
Add ability for Job class to define it's preferred queue(s)

**Rule:** First on the list is default, if not specified default to None.

This will be used for the default for execution at runtime as well as providing options for selection in UI.

Requires #2063.

### LINKED ISSUE #2065: Job Queue Routing: Add Job Model Definition for Preferred Queue(s)
Add ability for Job model to define it's preferred queue(s)

**Rule:** First on the list is default. If not specified, default to Job class's specification.

This will be used for the default for execution at runtime as well as providing options for selection in UI.

Requires #2064.

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes #2064, Closes #2065 
# What's Changed

Add new job model and class property named `task_queues` to present a list of supported queues for each job. This property defaults to an empty list which only allows the job to run on the default queue. The property may be defined on the class and can be overridden similar to other job properties like `read_only` and `approval_required`. The list of queues will be presented in the job run form to select the queue the job will be sent to. The queue can also optionally be specified in an api job run request, defaulting to the first queue in the list if the `worker_queue` parameter is not passed to the api call.

- Add setting `CELERY_TASK_DEFAULT_QUEUE` with a default value of `default` to nautobot config
- Add `task_queues` job class meta property
- Add `task_queues` job model field
- Add `task_queues_override` job model field
- Add new model fields to job edit form
- Add new model fields to job detail page
- Add task queue field to job run form
- Add `task_queue` field to job run API endpoint
- Add `task_queue` to `JobResult.job_kwargs` to store the queue for populating the form when clicking re-run on a job result
- Add example to example plugin jobs
- Update docs

## Job Run form

image

## Job Edit form

image

## Job Detail

image

# TODO

- [x] Explanation of Change(s)
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by jathanism — COMMENTED** [review]
`:%s/worker.queue/task.queue/g`
`:%s/Worker queue/Task queue/g`

File: examples/example_plugin/example_plugin/jobs.py [hunk 1]
```diff
@@ -34,6 +34,11 @@ class ExampleLoggingJob(Job):
     class Meta:
         name = "Example logging job."
         description = "I log stuff to demonstrate how UI logging works."
+        worker_queues = [
+            "",  # blank string is the default celery queue
```
[the remark this answers is not in the document]
Comment (glennmatthews) [reply examples/example_plugin/example_plugin/jobs.py:38]: Also, looking at screenshot https://user-images.githubusercontent.com/75227981/190172669-be31c371-ffc9-4c1f-bc25-d672f6422798.png, it looks like we have some self-inconsistency between `""` and `null` in various cases?

File: nautobot/extras/models/jobs.py [hunk 2]
```diff
@@ -170,6 +171,12 @@ class Job(PrimaryModel):
         help_text="Maximum runtime in seconds before the job will be forcibly terminated."
         "<br>Set to 0 to use Nautobot system default",
     )
+    worker_queues = JSONArrayField(
+        base_field=models.CharField(blank=True, null=True, max_length=100),
```
Comment (HanlinMiao) [inline nautobot/extras/models/jobs.py:175]: if we remove `null=True` here. Would that resolve the inconsistency between `""` and `null` in `worker_queues`?

**Review by bryanculver — APPROVED** [review]

File: nautobot/docs/configuration/optional-settings.md [hunk 3]
```diff
@@ -234,6 +234,16 @@ Celery broker URL used to tell workers where queues are located.
 
 ---
 
+## CELERY_TASK_DEFAULT_QUEUE
+
+Environment Variable: `NAUTOBOT_CELERY_TASK_DEFAULT_QUEUE`
+
+Default: `'default'`
```
Comment (glennmatthews) [inline nautobot/docs/configuration/optional-settings.md:241]: Is this a change from prior releases? It was `celery` before, right? Is there anything we need to call out in the release-notes about the upgrade changing this?

File: nautobot/extras/tests/test_views.py [hunk 4]
```diff
@@ -1515,6 +1515,8 @@ def setUpTestData(cls):
             "time_limit": 650,
             "has_sensitive_variables": False,
             "has_sensitive_variables_override": True,
+            "task_queues": "celery,priority",
```
Comment (glennmatthews) [inline nautobot/extras/tests/test_views.py:1518]: Should this be `default,priority`?

File: nautobot/core/settings.py [hunk 5]
```diff
@@ -649,6 +649,9 @@
 # Instruct celery to report the started status of a job, instead of just `pending`, `finished`, or `failed`
 CELERY_TASK_TRACK_STARTED = True
 
+# Default celery queue name that will be used by workers and tasks if no queue is specified
+CELERY_TASK_DEFAULT_QUEUE = os.getenv("NAUTOBOT_CELERY_TASK_DEFAULT_QUEUE", "default")
```
[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/core/settings.py:653]: I added a section in the release notes overview on this feature and it includes an `important` admonition calling out this changed setting.

File: nautobot/docs/configuration/optional-settings.md (hunk 3 above)
Comment (gsnider2195) [reply nautobot/docs/configuration/optional-settings.md:243]: I added a section to the release notes overview. See comment above

**Pull request merged**